### PR TITLE
Update dependencies and improve build system

### PR DESCRIPTION
**Pull request by fjorgemota**
Hey.

Here is my PR to update all the dependencies used by this package as I told you on Twitter.

The build size I was able to reach according to gzip-size is 4.04 kB, which IMHO is pretty decent. A few things that I think are good tasks to reduce the build size (which I wasn't able to do because I'm a bit busy here) is:

- Remove unused methods from `react-is`, which is a CommonJS module - Apparently rollup doesn't do the tree-shaking correctly here (probably because there's a call like `var reactIs = require("react-is");` in one of our dependencies);
- Remove `preact-context` in favour of a simpler context implementation;

This is still compatible with Preact 8, and it works both according to the unit tests and to my own tests on a personal project.

I hope you like it. :)

### CODE REVIEW

**Comment by fjorgemota** [comment]
Okay. After a few commits trying to resolve the build issues, now the build is passing.

I updated NodeJS to 10.x on Travis CI...I hope that's not a problem as it's the actual LTS version maintained.. :)

File: package.json [hunk 1]
```diff
@@ -45,54 +45,72 @@
...
-    "babel-plugin-transform-react-remove-prop-types": "^0.4.1",
-    "babel-preset-es2015": "^6.24.1",
-    "babel-preset-react": "^6.24.1",
-    "babel-preset-stage-0": "^6.24.1",
-    "chai": "^3.5.0",
-    "cross-env": "^4.0.0",
+    "@babel/cli": "^7.0.0",
+    "@babel/core": "^7.4.0",
+    "@babel/plugin-external-helpers": "^7.0.0",
+    "@babel/plugin-proposal-class-properties": "^7.4.0",
```
Comment (developit) [inline package.json]: What's the reason for inlining all of these? I think they're part of `@babel/preset-env`, right?

**Comment by developit** [comment]
Awesome! Thanks @fjorgemota! I would like to move this project over to using [microbundle](https://github.com/developit/microbundle), since it should cover pretty much everything this rollup configuration is doing at this point. If that's something you're interested in doing let me know, otherwise I'm happy to!

I'm also happy to add you as a contributor here.

File: package.json (hunk 1 above)
Comment (fjorgemota) [reply package.json]: Oh. They're not part of `@babel/preset-env` but they were not used by the config.

I removed most of these packages now and fixed a few things in babel config. Thanks for the observation. :)

**Comment by fjorgemota** [comment]
Oh. I think that's a good idea, but I'm not sure if this should be done already in this PR...in either way, as I don't have actual experience with `microbundle` I'm not sure if I could do these changes fast, as I'm a bit busy here too. 

I would love to help to contribute more updates with this package as I'll use it for a few more months and I think I can help with a few changes here and there. I think a version bump is very welcome, too. :)

**Comment by fjorgemota** [comment]
Hey.

I took a look at microbundle and tried to build this project with it...but I found an error (during the build), , which I already fixed with the custom rollup config I made...

Basically, to build this project with all the updated dependencies, [this specific config to rollup-commonjs plugin](https://github.com/developit/preact-redux/blob/ca54f35681d2c2bfb9216fc81dbd4ee7fe4e6369/rollup.config.js#L68-L70) is VERY needed, because it allows rollup to find the named exports for the react-is dependency and so it allows the project to be built.

Looking through microbundle's code, I found that [there's no way to pass custom options to the commonjs plugin](https://github.com/developit/microbundle/blob/master/src/index.js#L488-L491), so I don't think this project could use microbundle for now, as this very specific condition is quite needed. 

What do you think, @developit?

**Comment by developit** [comment]
Makes sense to me. I think we'll want to swap out react-is at some point, since it doesn't really work with Preact.

**Comment by fjorgemota** [comment]
Yeah, very probably. It's another change that could reduce more a few good bytes from the final bundle.

**Comment by fjorgemota** [comment]
Well, this is already good to merge?

**Comment by lauritzsh** [comment]
Is it possible to have this published to npm? Right now I'm dependent on a specific commit but it would be better to just specify a version in package.json.

**Pull request merged**